integration-tests/features/steps/stack_analysis.py

```diff
@@ -104,26 +110,43 @@ def perform_valid_manifest_post(context, manifest, url):
     print(response.json())
 
 
-def send_manifest_to_stack_analysis(context, manifest, name, endpoint, use_token, **kwargs):
+def send_manifest_to_stack_analysis(version, context, manifest, name, endpoint, use_token, **kwargs):
     """Send the selected manifest file to stack analysis."""
     filename = 'data/{manifest}'.format(manifest=manifest)
     manifest_file_dir = os.path.dirname(filename)
     path_to_manifest_file = os.path.abspath(manifest_file_dir)
     context.manifest = manifest
 
-    # please note that the trick with (None, path_to_manifest_file) has to be
-    # used here so the REST API call would work properly. It is similar to use
-    # curl -F 'manifest[]=@filename' -F 'filePath[]=PATH_TO_FILE'
-    files = {'manifest[]': (name, open(filename, 'rb')),
-             'filePath[]': (None, path_to_manifest_file)}
+    # Build request files, data and headers.
+    files = {}
+    data = {}
+    headers = {}
+    if version == 4:
+        # For stack analyses V2 we need only single manifest file and path is a string params.
+        files = {'manifest': (name, open(filename, 'rb'))}
+        data = {
+            'file_path': path_to_manifest_file,
+            'ecosystem': kwargs.get('ecosystem', None)
+        }
+        if use_token:
+            headers = authorization(context)
+    else:
+        # please note that the trick with (None, path_to_manifest_file) has to be
+        # used here so the REST API call would work properly. It is similar to use
+        # curl -F 'manifest[]=@filename' -F 'filePath[]=PATH_TO_FILE'
+        files = {'manifest[]': (name, open(filename, 'rb')),
+                 'filePath[]': (None, path_to_manifest_file)}
+        if use_token:
+            headers = authorization_with_eco_origin(context,
```

Comment:
using `authorization_with_eco_origin()` means you are using the OSIO auth please look at [https://github.com/fabric8-analytics/fabric8-analytics-common/blob/master/integration-tests/features/steps/three_scale_preview.py#L40](url) and  i would suggest you to make your functions or modifications in `three_scale_preview.py` file